src/modules/release/utils.ts

```diff
@@ -0,0 +1,373 @@
+import chalk from 'chalk'
+import {execSync} from 'child-process-es6-promise'
+import {
+  close,
+  existsSync,
+  openSync,
+  pathExistsSync,
+  readdir,
+  readFileSync,
+  writeSync
+} from 'fs-extra'
+import Github from 'github-api'
+import * as inquirer from 'inquirer'
+import {safeLoad} from 'js-yaml'
+import {dirname, join} from 'path'
+import {find, map, prop, replace} from 'ramda'
+import log from '../../logger'
+
+const yarnPath = require.resolve('yarn/bin/yarn')
+
+export const setup = (
+  versionProvider: any,
+  releaseType: string,
+  tagName: string
+) => {
+  let version = versionProvider.readVersion()
+  // Support for "promote" predicate, which bumps prerelease to stable,
+  // without changing version number.
+  if (releaseType === 'promote') {
+    // Promote only makes sense when there is a prerelease.
+    if (version.prerelease.length === 0) {
+      throw new Error(`The version you are trying to promote to stable (
+        ${version.format()}) is already stable.\n`)
+    } else {
+      version.prerelease = []
+      tagName = 'stable'
+    }
+  }
+  // For other types, simply increment
+  else {
+    const incrementType = releaseType === 'prerelease' ||
+      tagName === 'stable' ||
+      version.prerelease.length === 0
+      ? releaseType || 'patch'
+      : (`pre${releaseType}`) || 'prepatch'
```

Comment:
This is kind of illegible, too many expressions in the same assignment with no indentation

Reply:
Sorry this file was not supposed to be there.